lab_3/main.py

```diff
@@ -0,0 +1,300 @@
+import argparse
+import logging
+import os
+import csv
+import shutil
+from datetime import datetime, timedelta
+import urllib.request
+import pytz
+
+
+class UserData:
+    """ Class to work with user database """
+    def __init__(self):
+        """
+        Initializes an instance of UserData class.
+        This constructor sets up the argument parser and parses the command line arguments.
+        It also initializes various instance variables used for data processing.
+
+        :param None:
+        """
+        parser = argparse.ArgumentParser()
+        parser.add_argument('--destination', type=str, required=True)
+        parser.add_argument('--filename', type=str, default='output')
+        parser.add_argument('--sort')
+        parser.add_argument('--log_level', type=str)
+        self.args = parser.parse_args()
+        self.destination = f'{self.args.destination}{self.args.filename}.csv'
+        self.new_path = fr'{self.args.destination}\Data'
+        self.data = []
+        self.fieldnames = []
+        self.sorted_data = []
+        self.rearranged_data = {}
```

Comment:
I'm not sure, but please check, some of the parameters can be assigned as **None**